sympy/physics/optics/medium.py

```diff
@@ -152,6 +135,73 @@ def refractive_index(self):
         """
         return (c/self.speed)
 
+    def __lt__(self, other):
+        """
+        Compares based on refractive index of the medium.
+        """
+        return self.refractive_index < other.refractive_index
+
+    def __gt__(self, other):
+        return not self < other
+
+    def __eq__(self, other):
+        return self.refractive_index == other.refractive_index
+
+    def __ne__(self, other):
+        return not self == other
+
+
+class MediumN(Medium):
+
+    def __new__(cls, name, n):
+        obj = super(Medium, cls).__new__(cls, name, n)
+        obj.name = name
+        obj._n = n
+        return obj
+
+    @property
+    def n(self):
+        return self._n
+
+    def __str__(self):
+        return type(self).__name__ + ': ' + sstr([self.n])
```

Comment:
This `__str__` method should be removed. The printers control the printed display of Basic